File: laboratorios/lab05/codigo_estudiante/Java/AyudaEjercicioDeUVA/GraphAL.java
import java.util.ArrayList;
/**
 * Esta clase es una implementación de un grafo usando listas de adyacencia
 * 
 * @author Mauricio Toro 
 * @version 1
 */
public class GraphAL extends Graph
{
   private ArrayList<ArrayList<Integer>> vertices;
   private ArrayList<ArrayList<Integer>> weights;
   
   public GraphAL(int size)
   {
       super(size);
       vertices = new ArrayList<ArrayList<Integer>>();
       weights = new ArrayList<ArrayList<Integer>>();
       for (int i = 0; i < size; i++)
       {
        ArrayList<Integer> arcs = new ArrayList<Integer>();
        ArrayList<Integer> ws = new ArrayList<Integer>();
        vertices.add(arcs);
        weights.add(ws);
       }
   }
   
   public void addArc(int source, int destination, int weight)
   {
     
       vertices.get(source).add(destination);
       weights.get(source).add(weight);
       
       vertices.get(destination).add(source);
       weights.get(destination).add(weight);
   }
   
   public int getWeight(int source, int destination)
   {
      for (int j = 0; j < size; j++)
      {
        if (vertices.get(source).get(j) == destination)
            return weights.get(source).get(j);
      }
      return -1;
   }
  
   public ArrayList<Integer> getSuccessors(int vertice)
   {
      return vertices.get(vertice);
   }
}

File: talleres/taller11/spoiler/python/GraphAl.py
class GraphAl:
    
    def __init__(self, size):
        self.size = size
        self.lista = [[] for i in range(size)]

    def __repr__(self):
        return '{}'.format(self.lista)

    def getWeight(self, source, destination):
        for d in self.lista[____]:
            if d[___] == ____:
                return d[___]

    def addArc(self, source, destination, weight):
        self.lista[____].append((_____, ____))

    def getSuccessors(self, vertice):
        succs = []
        for d in self.lista[____]:
            succs.append(____)
        return succs

    


ga = GraphAl(3)
ga.addArc(0, 3, 10)
print(ga)
print(ga.getWeight(0, 3))
ga.addArc(0, 4, 7)
print(ga)
print(ga.getSuccessors(0))